src/common/imageio_png.c

```diff
@@ -229,16 +234,34 @@ int dt_imageio_png_read_profile(const char *filename, uint8_t **out)
 
   if(read_header(filename, &image) != 0) return DT_IMAGEIO_FILE_CORRUPTED;
 
+  /* TODO: also add check for known cICP chunk read support once added to libpng */
+#ifdef PNG_STORE_UNKNOWN_CHUNKS_SUPPORTED
+  png_unknown_chunkpp unknowns = NULL;
+  int num = png_get_unknown_chunks(image.png_ptr, image.info_ptr, unknowns);
+  for(size_t c = 0; c < num; ++c)
+    if(!strcmp((const char *)unknowns[c]->name, "cICP"))
+    {
+      /* only RGB and full range pixel values are supported by the loader above and dt color management */
+      if(!unknowns[c]->data[2] && unknowns[c]->data[3])
```

Comment:
Isn't there a missing ! above:

`   && !unknowns[c]->data[3]`

Reply:
Nope, we actually want the first one (matrix coeffs) to be 0 and the second (range) to be 1. I'll clarify with a comment.